src/GovUk.Education.ExploreEducationStatistics.Admin/Services/Public.Data/ProcessorClient.cs

```diff
@@ -0,0 +1,55 @@
+#nullable enable
+using GovUk.Education.ExploreEducationStatistics.Admin.Services.Interfaces.Public.Data;
+using GovUk.Education.ExploreEducationStatistics.Common.Extensions;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Requests;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.ViewModels;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Net.Http;
+using System.Net.Http.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace GovUk.Education.ExploreEducationStatistics.Admin.Services.Public.Data;
+
+internal class ProcessorClient(ILogger<ProcessorClient> logger, HttpClient httpClient) : IProcessorClient
+{
+    public async Task<Either<ActionResult, ProcessorTriggerResponseViewModel>> Process(
+        Guid releaseFileId, 
+        CancellationToken cancellationToken = default)
+    {
+        var request = new ProcessorTriggerRequest
+        {
+            ReleaseFileId = releaseFileId,
+        };
+
+        var response = await httpClient
+            .PostAsJsonAsync("api/orchestrators/processor", request, cancellationToken: cancellationToken);
+
+        if (!response.IsSuccessStatusCode)
+        {
+            switch (response.StatusCode)
+            {
+                default:
+                    var message = await response.Content.ReadAsStringAsync(cancellationToken);
+                    logger.LogError(
+                        StringExtensions.TrimIndent(
+                        $"""
+                         Failed to create data set version with status code: {response.StatusCode}. Message:
+                         {message}
+                         """));
+                    response.EnsureSuccessStatusCode();
```

Comment:
Appreciate that we discussed just throwing a server error, but does feel like this would lead to an unnecessarily bad user experience on the frontend.

I've mentioned this in various comments on #4825, but we could pipe validation errors from the processor endpoint through to the frontend so we can give better error feedback to the end user.

Reply:
I've changed this to now check for 400s and 404s, and it will bubble the `ValidationProblemDetails` from the 400s back to the caller.